subprojects/core-api/src/main/java/org/gradle/StartParameter.java

```diff
@@ -850,4 +851,24 @@ public boolean isInteractive() {
     public void setInteractive(boolean interactive) {
         this.interactive = interactive;
     }
+
+    /**
+     * Whether caching for native tasks is enabled.
+     *
+     * @since 4.3
+     */
+    @Incubating
+    public void setNativeCachingEnabled(boolean nativeCachingEnabled) {
+        this.nativeCachingEnabled = nativeCachingEnabled;
+    }
```

Comment:
Does this need to be public?

Reply:
My 2 cents, I think it would be better to have an undocumented, magic system property vs a property in StartParameter we remove later.

Reply:
@lptr What is your take on this? The magic system property can be set via the command line and via `gradle.properties`, too. It seems better to me since the flag would only be available for `4.3` anyway.